aea/skills/tasks.py

```diff
@@ -18,72 +18,102 @@
 
 """This module contains the classes for tasks."""
 import logging
+import multiprocessing
+import signal
 import threading
-from collections import deque
-from concurrent.futures import Executor, ThreadPoolExecutor
-from queue import Queue
-from threading import Thread
-from typing import Optional
-
-from aea.skills.base import Task
+from multiprocessing.pool import AsyncResult, Pool
+from typing import Any, Callable, Dict, Optional, Sequence, cast
 
 logger = logging.getLogger(__name__)
 
 
+def init_worker():
```

Comment:
this is needed to disable the signal handler for `SIGINT` signals from the main process to children processes.

Without this, there is a bad behaviour of the `KeyboardInterrupt` handler, which is a known issue.

Reply:
It prevents graceful teardown of tasks, but at least we guarantee the pool does not hang... it's a trade-off.